src/java/org/apache/cassandra/io/sstable/metadata/StatsMetadata.java

```diff
@@ -340,7 +345,27 @@ public void serialize(Version version, StatsMetadata component, DataOutputPlus o
         public StatsMetadata deserialize(Version version, DataInputPlus in) throws IOException
         {
             EstimatedHistogram partitionSizes = EstimatedHistogram.serializer.deserialize(in);
+            
+            if (partitionSizes.isOverflowed())
+            {
+                logger.warn("Deserialized partition size histogram with {} values greater than the maximum of {}. " +
```

Comment:
would this be candidate for no spam logger? I think it will mainly be on open() but might be worth considering (ok to ignore this)